plutus-tx/src/Language/PlutusTx/AssocMap.hs

```diff
@@ -76,6 +78,18 @@ lookup c (Map xs) =
         go ((c', i):xs') = if c' == c then Just i else go xs'
     in go xs
 
+{-# INLINABLE insert #-}
+insert :: forall k v . Eq k => k -> v -> Map k v -> Map k v
+insert k v m = unionWith (\_ b -> b) m (fromList [(k, v)])
```

Comment:
I saw you using this in a loop: you're going to get terrible quadratic performance doing that, FYI.

Reply:
I'm going to change the implementation to AVLTree/RBTree as soon as those land in, so I think it doesn't matter that much.